content/utils.js

```diff
@@ -0,0 +1,53 @@
+const path = require("path");
+
+const sanitizeFilename = require("sanitize-filename");
+
+function buildURL(locale, slug) {
+  if (!locale) throw new Error("locale falsy!");
+  if (!slug) throw new Error("slug falsy!");
+  return `/${locale}/docs/${slug}`.toLowerCase();
+}
+
+function slugToFoldername(slug) {
+  return (
+    slug
+      // We have slugs with these special characters that would be
+      // removed by the sanitizeFilename() function. What might then
+      // happen is that it leads to two *different slugs* becoming
+      // *same* folder name.
+      .replace(/\*/g, "_star_")
+      .replace(/::/g, "_doublecolon_")
+      .replace(/:/g, "_colon_")
+      .replace(/\?/g, "_question_")
+
+      .toLowerCase()
+      .split("/")
+      .map(sanitizeFilename)
+      .join(path.sep)
+  );
+}
+
+function memoizeDuringBuild(fn) {
+  if (process.env.NODE_ENV === "development") {
+    return fn;
+  }
+
+  const cache = new Map();
+  return (...args) => {
+    const key = JSON.stringify(args);
```

Comment:
If multiple calls to `fn` include objects in `args` but don't use the same key order for those objects, then the memoization will be replicated. It might not be worth making this more robust in its handling of objects, but just something to keep in mind

Reply:
Yup, good point! I also considered explicitly having a cache key getter as a parameter, but the call sites were simple enough that it seemed fine. I'll add a comment though, explaining the limitation.